common/density/rectangular_prism.h

```diff
@@ -0,0 +1,33 @@
+/* 
+ * File:   rectangular_prism.h
+ * Author: patrick
+ *
+ * Created on February 16, 2015, 4:20 PM
+ */
+
+#ifndef RECTANGULAR_PRISM_H
+#define	RECTANGULAR_PRISM_H
+
+class RectangularPrism
```

Comment:
I do not object to this code.  I will opine for a moment on matters of style, give an alternative or two (we'll see) and in the end say that this is fine.

I like std::pair more than most and it has some excellent uses.  Most of them are too clever by far and improving readability is rarely one of std::pairs virtues.  The problem is that it is not always obvious what is in first and what is in second.  This is compounded if they are of the same type.  I find that if there are good names for the components of a tuple (such as x,y,z for a point) it is almost always preferable to create a structure with named members.

Now I also agree that the getter/setter syntax is heavy-handed and can be avoided entirely with something like this:

struct RectangularPrisim
{
RectangularPrisim(int x1,int y1,...):x(x1),y(y1)..{/*Constructor exactly as you have it}

const int x;
const int y;
/*etc*/
}

Using constant members with public access is equivalant and I think reads a bit better.   Now if there were setters also, and there was any possiblity of the internal state being more complex, or needing to maintain some class invariant then getters/setters are the way to go, and there is really no good syntax for them in c++ nor reasonable conventions to follow.  As such I'd leave the code as it is.